homeassistant/components/media_player/__init__.py

```diff
@@ -136,12 +136,11 @@
 
 ENTITY_ID_FORMAT = DOMAIN + ".{}"
 
-CACHE_IMAGES = "images"
-CACHE_MAXSIZE = "maxsize"
-CACHE_LOCK = "lock"
-CACHE_URL = "url"
-CACHE_CONTENT = "content"
-ENTITY_IMAGE_CACHE = {CACHE_IMAGES: collections.OrderedDict(), CACHE_MAXSIZE: 16}
```

Comment:
But annoying that we're creating a type for one-off use. I would expect adding `: Final` here it should be able to interfere the type?

Reply:
I couldn't find a cleaner / prettier way to do it. I'm happy to leave open to wait for suggestions...

Reply:
> But annoying that we're creating a type for one-off use. I would expect adding `: Final` here it should be able to interfere the type?

Adding `Final` does help with type inference but it doesn't change a `dict` to a `TypedDict`. I.e. with `Final` the type would be `dict[str, OrderedDict | int]`.

Using a `TypedDict` is a bit more work, but the correct choice here if it should be typed.